pymc3/tests/test_distributions_random.py

```diff
@@ -539,6 +539,23 @@ def test_probability_vector_shape(self):
         assert pm.Categorical.dist(p=p).random(size=4).shape == (4, 3, 7)
 
 
+class TestDirichlet(SeededTest):
+    @pytest.mark.parametrize(
+        "shape, size, out_shape",
+        [
+            ((2), (1), (1, 2)),
+            ((2), (2), (2, 2)),
+            ((2, 2), (2, 100), (2, 100, 2, 2)),
+            ((3, 4), (3, 4), (3, 4, 3, 4)),
+            ((3, 4), (3, 4, 100), (3, 4, 100, 3, 4)),
+            ((3, 4), (100), (100, 3, 4)),
+            ((3, 4), (1), (1, 3, 4)),
+        ],
+    )
+    def test_dirichlet_random_shape(self, shape, size, out_shape):
+        assert pm.Dirichlet.dist(a=np.ones(shape)).random(size=size).shape == out_shape
```

Comment:
The `out_shape` can be simplified to match how tests for other distributions are framed. Something like

```python
out_shape = to_tuple(size) + to_tuple(shape)
```